src/Transport/RestTransport.php

```diff
@@ -100,7 +100,7 @@ public static function build(string $apiEndpoint, string $restConfigPath, array
         $requestBuilder = $config['hasEmulator']
             ? new InsecureRequestBuilder("$baseUri:$port", $restConfigPath)
             : new RequestBuilder("$baseUri:$port", $restConfigPath);
-        $httpHandler = $config['httpHandler'] ?: self::buildHttpHandlerAsync();
+        $httpHandler = $config['httpHandler'] ?: self::buildHttpHandlerAsync($config['logger'] ?? null);
```

Comment:
add `logger => null` to the `$config` array above
```suggestion
        $httpHandler = $config['httpHandler'] ?: self::buildHttpHandlerAsync($config['logger']);
```